bin/reth/src/utils.rs

```diff
@@ -108,18 +111,65 @@ impl<'a, DB: Database> DbTool<'a, DB> {
         skip: usize,
         len: usize,
         reverse: bool,
-    ) -> Result<Vec<(T::Key, T::Value)>> {
+        search: Vec<u8>,
+        only_count: bool,
```

Comment:
I'm a bit lost here,

what do all these parameters do?
we don't have docs, and no idea what search is supposed to be

I'd like to move these two params into a helper struct, like `ListFilter` or something